File: ch5/CryptoAccount.py
#
# From: Blockchain Tech Notes by P. G. Bradford
#

class CryptoAccount(object):

    class Instance:
        publicKey = None
        secretKey= None
        dateTime  = None
        expiration = None
        block  = None
        amount = None
    
    def __init__(self,publicKey, secretKey, dateTime, expiration, block, amount):
        self.instance = self.Instance()
        self.instance.publicKey = publicKey
        self.instance.secretKey = secretKey
        self.instance.dateTime = dateTime
        self.instance.expiration = expiration
        self.instance.block = block
        self.instance.amount = amount

File: cpp/Ch5/CryptoAccount.cpp
#include <string>
#include <ctime>

class CryptoAccount {
public:
    struct Instance {
        std::string publicKey;
        std::string secretKey;
        std::time_t dateTime;
        std::time_t expiration;
        int block;
        double amount;
    };

    CryptoAccount(std::string publicKey, std::string secretKey, std::time_t dateTime, std::time_t expiration, int block, double amount) {
        instance.publicKey = publicKey;
        instance.secretKey = secretKey;
        instance.dateTime = dateTime;
        instance.expiration = expiration;
        instance.block = block;
        instance.amount = amount;
    }

private:
    Instance instance;
};

